app/src/org/commcare/dalvik/activities/LoginActivity.java

```diff
@@ -97,11 +121,14 @@ public void onCreate(Bundle savedInstanceState) {
                 password.requestFocus();
             }
         }
+
+        final Activity loginAct = this;
         
         login.setOnClickListener(new OnClickListener() {
 
             public void onClick(View arg0) {
                 errorBox.setVisibility(View.GONE);
+                ViewUtil.hideVirtualKeyboard(loginAct);

```

Comment:
You can use the pattern

LoginActivity.this
to get the pointer to the outer class here, which will remove the need for the final declartion above